process/smartContract/process.go

```diff
@@ -46,6 +46,9 @@ type scProcessor struct {
 	txTypeHandler process.TxTypeHandler
 	gasHandler    process.GasHandler
 	gasCost       GasCost
+	// This is set when the deployment or execution fails for some reason,
+	// but the originating transaction isn't supposed to be treated as erroneous
+	lastSilentError error
```

Comment:
:+1: This seems very useful when debugging a test with SC value queries, which don't return errors.